### PR TITLE
Docker support

### PR DESCRIPTION
**Pull request by sbchapin**
This PR fixes up the Docker usage of the ejected template.  It appears that the docker support for the native packaging was broken after #7 and very likely after #6 .  But no longer after this.

## Two big things in this PR:
1. CI-supported [Docker builds & runs of the native image](https://github.com/sbchapin/quickstart-scala-sbt-native-image.g8/runs/4379761927?check_suite_focus=true)
2. Replacement of `sbt-native-packager` plugin with `sbt-native-image`.  It...
    - slims down usage,
    - eliminates weird docker-in-docker possibilities, and...
    - offers conveniences like automatically downloading Graal & native-image via coursier.

### CODE REVIEW

File: src/main/g8/README.md [hunk 1]
```diff
@@ -89,19 +89,28 @@ Below are the libraries used to provide a broad starting base for this project.
...
+`sbt-native-image` is an sbt plugin that allows sbt to build assembled JRE-less executables using GraalVM.  This plugin handles downloading and installation of the related Graal JDK and native-image tool which are required to make the executable binary.  However, these are very particular about their targets - That's where we pick up with docker in this repo, ultimately ending up with a portable, fast, minified docker image with a single binary in it:
+
+1. **sbt + sbt-native-image + GraalVM + native-image** get us from **scala-to-bytecode-to-llvm-to-libc**.
+2. **docker** gets us from **libc-to-kernel**, providing a consistent "target" for GraalVM & native-image.
+
+However, it isn't all roses.  **The native-image & AoT aspect of any project building with Graal is very brittle**, as all classes exposed exclusively via reflection at run-time [must be explicitly provided to Graal](https://www.graalvm.org/reference-manual/native-image/Reflection/), whether determined automatically or manually.  
+
+Do your reading before you decide to push the outputs of native-image to production, and consider the cost of maintaining reflection mappings.
```
Comment (SwiftEngineer) [inline src/main/g8/README.md]: > consider the cost of maintaining reflection mappings

If it were only reflection mappings I'd be stoked. For me, most often it's the substitutions and random LLVM shit that really grinds my goats (getting segfaults because dns lookups require dynamically linked libc [for example](https://github.com/oracle/graal/issues/571)). 

I'm only mentioning it because (assuming you're crazy enough to use it, I always advice folks to avoid it) graal native's tracing agent does as pretty dang good job of making all your reflection problems disappear. So if you give folks the impression that reflection is the only issue, they might get a false sense of security. Though, I suppose if they were to truly heed your warning at the start of that sentence ("Do your reading"), they might figure that out for themselves.

That all being said, I ain't saying I know how to make an effective warning sign for graal headaches either. I think [the disclaimer](https://carpe.github.io/scalambda/docs/thegraaaaal/#disclaimers) I used for Scalambda was far too wordy. So much so that I wouldn't be surprised if anyone has ever actually read it.

File: src/main/g8/README.md [hunk 2]
```diff
@@ -89,19 +89,28 @@ Below are the libraries used to provide a broad starting base for this project.
...
 
 ###### Why do we use it? ######
 
-- GraalVM integration allows us to build small fast-starting binaries.
-- Bake dependencies into binaries.
+- GraalVM integration allows us to build small fast-starting binaries out of Scala that would have otherwise been bytecode.  Throwing the JRE baby out with the bathwater offers several benefits:
+  - No JVM cold start
+  - AoT target-based compilation means no JIT jitters or hotspot hotflashes
+  - Tiny docker images (tens of MB) with the smallest possible base image *(even from [scratch](https://hub.docker.com/_/scratch), if you want)*
```
Comment (SwiftEngineer) [inline src/main/g8/README.md:108]: Pretty sure I even had a 6MB image one time, which was pretty freakin' crazy.

File: src/main/g8/Dockerfile [hunk 3]
```diff
@@ -15,11 +15,13 @@
...
 #
-# Use a dockerized native builder
+# 1. Use a dockerized native builder and 2. Let sbt's coursier know it doesn't need to install GraalVM
 FROM ghcr.io/graalvm/graalvm-ce:java11-21.3.0 as binarybuilder
+ENV GRAALVM_INSTALLED=true
 
-# Install native-image so we can emit a pre-built statically-linked binary
+# 1. Install native-image so we can emit a pre-built statically-linked binary and 2. Let sbt know it doesn't need to install native-image
 RUN gu install native-image
+ENV NATIVE_IMAGE_INSTALLED=true
```
Comment (SwiftEngineer) [inline src/main/g8/Dockerfile:24]: that's pretty sweet that it handles that 👍

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -102,3 +98,62 @@ jobs:
...
+
+      - name: docker run "hi" program
+        working-directory: ./target/g8
+        run: |
+          docker run --rm sbchapin/quickstart-scala-sbt-native-image-example:test hi
+
+      - name: docker run "pi" program
+        working-directory: ./target/g8
+        run: |
+          docker run --rm sbchapin/quickstart-scala-sbt-native-image-example:test pi
```
Comment (SwiftEngineer) [inline .github/workflows/ci.yml:159]: Very nice!

**Review by SwiftEngineer — APPROVED** [review]
Sweeeeeet 💯 💯 💯

File: src/main/g8/README.md (hunk 1 above)
Comment (sbchapin) [reply src/main/g8/README.md]: I'm gonna add those comments to this README. Maybe it saves someone.

File: src/main/g8/README.md (hunk 2 above)
Comment (sbchapin) [reply src/main/g8/README.md:108]: Yikes. I'd hate to shell into that badboy.
